packages/synapse-react-client/src/components/ChallengeSubmission/ChallengeSubmissionStepper.tsx

```
@@ -0,0 +1,243 @@
+      ? steps.SELECT_COMMIT
+      : steps.SELECT_EVALUATION
+  const [step, setStep] = useState<Step>(initialStep)
+  const [errorMessage, setErrorMessage] = useState<string>()
+  const [selectedCommit, setSelectedCommit] = useState<DockerCommit>()
+
+  const [submissionName, setSubmissionName] = useState<string>('')
+  const [selectedEval, setSelectedEval] = useState<string | undefined>()
+  const [confirming, setConfirming] = useState<boolean>(false)
+
+  const hide = () => {
+    setErrorMessage(undefined)
+    setSelectedCommit(undefined)
+    setSubmissionName('')
+    setSelectedEval(undefined)
+    setConfirming(false)
+    onClose()
+    setStep(initialStep)
+  }
+
+  function handleStepChange(value?: StepsEnum) {
+    if (!value || !steps[value]) return
+    setErrorMessage(undefined)
+    setStep(steps[value])
+  }
+
+  async function confirmEligibility() {
+    if (!selectedEval) return setErrorMessage('Please select an evaluation.')
+    try {
+      return await SynapseClient.getSubmissionEligibility(
+        selectedEval,
+        teamId,
+        accessToken,
+      )
+    } catch (err) {
+      return setErrorMessage(err.message)
+    }
+  }
+
+  async function submitForEvaluation(eligibility: TeamSubmissionEligibility) {
+    if (
+      !entity.id ||
+      (entityType === EntityType.DOCKER_REPO && !selectedCommit)
+    )
+      return setErrorMessage('Error: Invalid entity or commit.')
+
+    const contributors = eligibility.membersEligibility
+      .filter(
+        member =>
+          !member.hasConflictingSubmission &&
+          member.isEligible &&
+          !member.isQuotaFilled &&
+          member.isRegistered,
+      )
+      .map(member => {
+        return { principalId: member.principalId.toString() }
+      })
+
+    const submission: EvaluationSubmission = {
+      userId: userId,
+      evaluationId: selectedEval!,
+      entityId: entity.id,
+      versionNumber: entity.versionNumber ?? 1,
```

@jay-hodgson Is this what you're looking for?

---

Yes!  I must have been looking at an old commit.  
Is entity.versionNumber generally defined here?  Still recommend to fallback to an invalid number (like -1) if this must be set, so it will show an error if it's attempting to use the fallback